### DISCUSSION TITLE
Other logs are not logged after NoSuchFileException during OperationLog.close() when restarting kyuubi-server in K8S env

### DISCUSSION THREAD

**Discussion by minyk** [apache/kyuubi#5421] [Q&A] [answered]
Hi, all

In my cases, `OperationLog.close()` try to close BufferedReader, but it says file is missing(NoSuchFileException). 
https://github.com/apache/kyuubi/blob/4bb67bdf3bcabd135102bd85c8992e9b52ba9356/kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala#L242

This happened after restarting kyuubi-server and PV is not set for `kyuubi.operation.log.dir.root`.

Errors are very similar like https://github.com/apache/kyuubi/pull/4986, but happens when reader.close() not delete

Also, when this happened, all other server logs are muted.

I try to quick fix to ignore NoSuchFileException in the `trySafely()`: 
```scala
  private def trySafely(f: => Unit): Unit = {
    try {
      f
    } catch {
      case _: NoSuchFileException =>
      case e: IOException =>
        // Printing log here may cause a deadlock. The lock order of OperationLog.write
        // is RootLogger -> LogDivertAppender -> OperationLog. If printing log here, the
        // lock order is OperationLog -> RootLogger. So the exception is thrown and
        // processing at the invocations
        throw new IOException(
          s"Failed to remove corresponding log file of operation: ${path.toAbsolutePath}",
          e)
    }
  }
```

This fix works for me, but I think more reliable way.

Thank you.

**Comment by yaooqinn** [comment]
Hi @minyk, this change looks promising, would you please file a Pull request?

**Reply by minyk** [reply, marked as answer]
@yaooqinn ok, I'll make the issue and PR. Thank you

**Reply by yaooqinn** [reply]
thank you
